src/js/App/GlobalFilter/GlobalFilterMenu.js

```diff
@@ -0,0 +1,169 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { TextInput, MenuList, MenuItem, Select, SelectVariant, MenuGroup, Checkbox } from '@patternfly/react-core';
+
+import './global-filter-menu.scss';
+
+export const groupType = {
+  checkbox: 'checkbox',
+  radio: 'radio',
+  button: 'button',
+  plain: 'plain',
+};
+
+const getMenuItems = (groups, filterValueRegex, onChange, calculateSelected) => {
+  const result = groups.map(({ value, label, id, type, items, ...group }) => ({
+    label,
+    value,
+    items: items
+      .filter(
+        (item) =>
+          group.noFilter ||
+          (value && filterValueRegex.test(value)) ||
+          (label && filterValueRegex.test(label)) ||
+          (item.value && filterValueRegex.test(item.value)) ||
+          (item.label && filterValueRegex.test(item.label))
+      )
```

Comment:
The `filter` from original implementation was there because Select required some filtering in the past, it wasn't working without it. Can you try it without this filter? The component will shrink even more.